Feature/farms search

memacius32 opened the pull request:
- Adding Search feature for pools page, Filtering by token or network name, or address. 
- Adding pools sorting by apr or liquidity.
- Adding render optimizations.

specialist0x commented on src/utils/pools.ts:
```diff
@@ -0,0 +1,34 @@
+  return token0.equals(TRI[token0.chainId]) ? token0IsFirst : token1IsFirst
+}
+
+export const poolIsStaking = (stakedAmount: TokenAmount | null) => {
```
This shouldn't be called `poolIsStaking` bc that implies it has context of the user
```suggestion
export const isTokenAmountPositive = (stakedAmount: TokenAmount | null) => {
```

specialist0x commented on src/pages/EarnTri/EarnTri.styles.tsx:
```diff
@@ -0,0 +1,99 @@
+  `};
+`
+
+export const StyledSortContainer = styled.div``
```
Is this needed?

memacius32 replied:
nope, leftover, cleaning 🧹

specialist0x commented on src/pages/EarnTri/index.tsx:
```diff
@@ -1,143 +1,198 @@
+  const [sortDescending, setSortDescending] = useState<boolean>(true)
+
+  const getSortedFarms = () => {
+    switch (sortBy) {
+      case SortingType.default:
+        return POOLS_ORDER.map(index => allFarmArrs[index])
+    }
+    switch (sortBy) {
+      case SortingType.liquidity:
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
+    }
+    switch (sortBy) {
+      case SortingType.totalApr:
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
+    }
```
Was there a reason for multiple `switch`s?

```suggestion
    switch (sortBy) {
      case SortingType.default:
        return POOLS_ORDER.map(index => allFarmArrs[index])
      case SortingType.liquidity:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
      case SortingType.totalApr:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
    }
```

memacius32 replied:
messed up!

specialist0x commented:
Works great!

Couple things:

1. Looks like Legacy Farms still shows, even if I'm not deposited:

![image](https://user-images.githubusercontent.com/94581898/149585975-6bb9f284-f0bc-4a7d-a4af-bcd3e58f9490.png)

2. on `EarnTri/index.tsx`, can you also add support for LP addresses?  Should be similar to your existing implementation with the addition of `allFarmArrs.map(v => v.lpAddress)`

3. I pushed up https://github.com/trisolaris-labs/interface/pull/128, make sure to handle the conflict or rebase ahead of it

memacius32 commented:
Rebased and fixed. 

Also I was wondering, do you agree it would make sense to filter addresses only with by exact match? I don't imagine anyone manually typing addresses
 Also our Swap and Farm search inputs say "...or paste address" in the placeholder

specialist0x commented:
> Also I was wondering, do you agree it would make sense to filter addresses only with by exact match?

Well, partial matches might be used? They'd also be able to see the card to visually verify

> Also our Swap and Farm search inputs say "...or paste address" in the placeholder
I think `Search farms` or `Search` works

memacius32 commented:
Alright, I leave the current functionality then, it filters with partial matches.

About the placeholder, what about "Search farms by name or address" ?🤔

specialist0x commented:
> About the placeholder, what about "Search farms by name or address" ?

Let's do `Search farms by token or address`

memacius32 commented:
@specialist0x done
